src/section/section_keys.rs

```diff
@@ -31,7 +35,7 @@ pub struct SectionKeysProvider {
     /// Our current section BLS keys.
     current: Option<SectionKeyShare>,
     /// The new keys to use when section update completes.
-    pending: BTreeMap<u64, DkgResult>,
+    pending: BTreeMap<DkgKey, DkgResult>,
```

Comment:
Not quite sure why this change is necessary. In what case would we get multiple completed DKGs for the same key index but different participants?

Reply:
previously, we start DKG whenever promote_demote_elders returns with a new_elder_info , which means multiple DKG with different participants with same section_index.
However, this may no longer be the case as there is your more fixing commits.
I will double check this once cherry-picked your work into PR 2188.

Reply:
This is also fixed by forcing name change on promote/demote. We really should do this to become crdt complient in any case.